LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -849,7 +849,7 @@ protected virtual async Task<LoRaADRResult> PerformADR(LoRaRequest request, LoRa
             if (!isFrameCounterFromNewlyStartedDevice && payloadFcnt <= loRaDevice.FCntUp)
             {
                 // most probably we did not ack in time before or device lost the ack packet so we should continue but not send the msg to iothub
-                if (payload.RequiresConfirmation && payloadFcnt == loRaDevice.FCntUp)
+                if (payload.RequiresConfirmation && payloadFcnt == loRaDevice.FCntUp && (concentratorDeduplicationResult is ConcentratorDeduplicationResult.NotDuplicate or ConcentratorDeduplicationResult.DuplicateDueToResubmission))
```

Comment:
@p-schuler we added the `ConcentratorDeduplicationResult.NotDuplicate` case in the condition to cover for the case of a late resubmission (when the cache entry is expired or the LNS crashed)

Reply:
Makes sense!